moa/src/main/java/moa/classifiers/lazy/kNN.java

```diff
@@ -40,12 +44,17 @@
  * @author Jesse Read (jesse@tsc.uc3m.es)
  * @version 03.2012
  */
-public class kNN extends AbstractClassifier implements MultiClassClassifier {
+public class kNN extends AbstractClassifier implements MultiClassClassifier, Regressor {
 
     private static final long serialVersionUID = 1L;
 
 	public IntOption kOption = new IntOption( "k", 'k', "The number of neighbors", 10, 1, Integer.MAX_VALUE);
 
+	// For checking if users want to predict with regression
+	public FlagOption regressionOption = new FlagOption("regression",'r',"Regression or not");
```

Comment:
You can remove this option and just check whether the class is numeric to choose between regression/classification.